man/adoc/bpftrace.adoc

```diff
@@ -0,0 +1,1900 @@
+[16M, 32M)            16 |                                                    |
+[32M, 64M)            27 |                                                    |
+[64M, 128M)           48 |@                                                   |
+[128M, 256M)          98 |@@@                                                 |
+[256M, 512M)         191 |@@@@@@                                              |
+[512M, 1G)           394 |@@@@@@@@@@@@@                                       |
+[1G, 2G)             820 |@@@@@@@@@@@@@@@@@@@@@@@@@@@                         |
+----
+
+=== lhist(int64 n, int64 min, int64 max, int64 step)
+
+Create a linear histogram of `n`.
+`lhist` creates `M` (`(max - min) / step`) buckets in the range `[min,max)` where each bucket is `step` in size.
+Values in the range `(-inf, min)` and `(max, inf)` get their get their own bucket too, bringing the total amount of buckets created to `M+2`.
+
+----
+i:ms:1 {
+  @ = lhist(rand %10, 0, 10, 1);
+}
+
+i:s:5 {
+  exit();
+}
+----
+
+Prints:
+
+----
+@:
+[0, 1)               306 |@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@         |
+[1, 2)               284 |@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@            |
+[2, 3)               294 |@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@          |
+[3, 4)               318 |@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@       |
+[4, 5)               311 |@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@        |
+[5, 6)               362 |@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@|
+[6, 7)               336 |@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@    |
+[7, 8)               326 |@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@      |
+[8, 9)               328 |@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@     |
+[9, 10)              318 |@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@       |
+----
+
+=== max(int64 n)
+
+Update the map with `n` if `n` is bigger than the current value held.
+
+=== min(int64 n)
+
+Update the map with `n` if `n` is smaller than the current value held.
+
+=== stats(int64 n)
+
+`stats` combines the `count`, `avg` and `sum` calls into one.
+
+----
+kprobe:vfs_read {
+  @bytes[comm] = stats(arg2);
+}
+----
+
+----
+@bytes[bash]: count 7, average 1, total 7
+@bytes[sleep]: count 5, average 832, total 4160
+@bytes[ls]: count 7, average 886, total 6208
+@
+----
+
+=== sum(int64 n)
+
+Calculate the sum of all `n` passed.
+
+=== zero(map m)
+
+*async*
+
+Set all values for all keys to zero.
+
+== Functions
+
+Functions that are marked *async* are asynchronous which can lead to unexpected behaviour, see the 'sync and async' section for more information. // TODO: LINK
+
+*compile time* functions are evaluated at compile time, a static value will be compiled into the program.
+
+*unsafe* functions can have dangerous side effects and should be used with care, the `--unsafe` flag is required for use.
+
+=== buf_t buf(void * data, [int64 length])
```

Comment:
this seems to break cross references. I think doing

```
=== buf

.variants
* `buf_t buf...
```

etc for all functions is better